interactions/api/http/guild.py

```diff
@@ -385,6 +385,20 @@ async def get_all_roles(self, guild_id: int) -> List[dict]:
 
         return request
 
+    async def get_role(self, guild_id: int, role_id: int) -> dict:
+        """
+        Gets a role from a Guild.
+
+        :param guild_id: Guild ID snowflake
+        :param role_id: Role ID snowflake
+        :return: A Role object as a dictionary.
+        """
+        request = await self.get_all_roles(guild_id)
+
+        for role in request:
+            if role.get("id") and int(role["id"]) == role_id:
+                return role
+
```

Comment:
This... this I feel iffy about.

There's clearly cases where some HTTP methods do some pre-processing (`get_user` comes to mind immediately), but from what I can see, there's *never* post-processing. At most, the client may cache the result, but I've never seen it explicitly filter things out like this. There's also no precedent for any other similar library to do something like this, either.

Worst of all, this doesn't even represent a valid Discord endpoint. While it would be admittedly nice to have a `Get Role` method, Discord explicitly doesn't have it, and we shouldn't be giving users the wrong expectation.
(You can argue aliases for HTTP methods give users the wrong expectations, but at least it's to a much lesser extent.)

This *would* make it easier for the future `bot.get`, and I recognize that, but I personally don't like this as a "solution" to that problem

Reply:
Huh, fl0w/James *has* stated this would be fine, [as seen here.](https://github.com/interactions-py/library/pull/702#discussion_r842243753)

I disagree, but that's something to consider.

Reply:
The previous solution (that my previous get method involved) was to make a conditional to check if it is a `Role` and do it manually there, which is *not* dynamic.
Resolving this outlier simplifies the `get()` method and improves consistency.

I don't know why Discord doesn't have a Get Role endpoint, so I had to use post-processing.
I don't know of a better solution than this, please let me know how you would go about this.
I have modified the description to `Gets the specified role from the list of roles in a Guild.` Should I also add that it isn't an endpoint to further clarify what it is?

Reply:
I *do* have an idea in mind, but it was shot down earlier. I may re-mention it again sooner or later, but probably not here. If James/fl0w wanted it, I can't argue about it too much.

Regardless, I do think it's worth clarifying it's not a native endpoint, yeah.

Reply:
Okay, will add, thanks